server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -338,130 +342,427 @@ describe("Routerlicious", () => {
+								"User 2 should have received 1 signal",
+							);
+							assert.equal(
+								user2Signal.content,
+								"TestSignal",
+								"User 2 signal content mismatch",
+							);
+							assert.equal(userSignalCounts[2], 1),
+								"User 3 should have received 1 signal";
+							assert.equal(
+								user3Signal.content,
+								"TestSignal",
+								"User 3 signal content mismatch",
+							);
+						});
+
+						// These type tests can be skipped to save time
+						describe("Invalid/Malformed signals", () => {
+							let noSignalPromises: Promise<void>[];
+
+							before(() => {
+								noSignalPromises = [
+									new Promise<void>((resolve) => {
+										firstSocket.on("signal", (signal: ISignalMessage) => {
+											assert.fail("Signal should not have been received");
+										});
+										setTimeout(resolve, 100);
+									}),
+									new Promise<void>((resolve) => {
+										secondSocket.on("signal", (signal: ISignalMessage) => {
+											assert.fail("Signal should not have been received");
+										});
+										setTimeout(resolve, 100);
+									}),
+									new Promise<void>((resolve) => {
+										thirdSocket.on("signal", (signal: ISignalMessage) => {
+											assert.fail("Signal should not have been received");
+										});
+										setTimeout(resolve, 100);
+									}),
+								];
+							});
+
+							it("Invalid targetClientID", async () => {
+								const targetedSignal: ISentSignalMessage = {
+									targetClientId: "invalidClientID",
+									content: "TestSignal",
+								};
+
+								secondSocket.send("submitSignal", secondConnectMessage.clientId, [
+									targetedSignal,
+								]);
+								await Promise.all(noSignalPromises);
+							});
+
+							it("Invalid targetClientID type", async () => {
+								const targetedSignal = {
+									targetClientId: true,
+									content: "TestSignal",
+								};
+
+								secondSocket.send("submitSignal", secondConnectMessage.clientId, [
+									targetedSignal,
+								]);
+								await Promise.all(noSignalPromises);
+							});
+							it("Missing content field", async () => {
+								const targetedSignal = {
+									targetClientId: firstConnectMessage.clientId,
+								};
+								secondSocket.send("submitSignal", secondConnectMessage.clientId, [
+									targetedSignal,
+								]);
+								await Promise.all(noSignalPromises);
+							});
+							it("Invalid signal field", async () => {
+								const targetedSignal = {
+									targetClientId: firstConnectMessage.clientId,
+									content: "TestSignal",
+									invalidField: "invalid",
+								};
+								secondSocket.send("submitSignal", secondConnectMessage.clientId, [
+									targetedSignal,
+								]);
+								await Promise.all(noSignalPromises);
+							});
+							it("Invalid optional signal fields", async () => {
+								const targetedSignal = {
+									targetClientId: firstConnectMessage.clientId,
+									content: "TestSignal",
```

Comment:
Can the content be json? If so, can it be malformed json?

Reply:
`content` can be anything, since it is typed as unknown and the server does not operate on it. Whatever it receives it will transmit and this is the expected behavior.

Reply:
I will add test cases with different content types to validate this ^